server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/authorization/templates_controller.rb

```diff
@@ -0,0 +1,67 @@
+##########################################################################
+# Copyright 2017 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  module Admin
+    module Authorization
+      class TemplatesController < BaseController
+        CLONER = Cloner.new
+
+        before_action :load_template, only: [:show, :update]
+        before_action :check_admin_user_and_401
+        before_action :check_for_stale_request, only: [:update]
+
+        def show
+          json = ApiV1::Admin::Authorization::AuthorizationConfigRepresenter.new(@template.getAuthorization).to_hash(url_builder: self)
+          render DEFAULT_FORMAT => json if stale?(etag: etag_for(@template))
+        end
+
+        def update
+          result = HttpLocalizedOperationResult.new
+          authorization = ApiV1::Admin::Authorization::AuthorizationConfigRepresenter.new(com.thoughtworks.go.config.Authorization.new).from_hash(params[:template])
+          updated_template = CLONER.deepClone(@template)
```

Comment:
How bout getting rid of the template cloning? 
change `template_config_service.updateTemplateAuthConfig(current_user, template_name, auth_config, result, etag)'
and UpdateTemplateAuthConfigCommand update can update the auth_config as part of the update method.

What do you think?